lms/djangoapps/discussion_api/tests/test_api.py

```diff
@@ -100,29 +104,36 @@ def get_course_topics(self, user=None):
         Get course topics for self.course, using the given user or self.user if
         not provided, and generating absolute URIs with a test scheme/host.
         """
-        return get_course_topics(self.course.id, user or self.user)
+        if user:
+            self.request.user = user
+        return get_course_topics(self.request, self.course.id)
 
-    def make_expected_tree(self, topic_id, name, children=None):
+    def make_expected_tree(self, topic_id_list, name, children=None):
```

Comment:
Instead of changing to topic_id_list, this should get the topic ids from the children